src/lib/structures/Josh.ts

```diff
@@ -0,0 +1,1813 @@
+    else {
+      payload.path = this.getPath(pathOrHook);
+      payload.value = value;
+    }
+
+    for (const middleware of this.middlewares.array()) await middleware.run(payload);
+    for (const middleware of this.middlewares.getPreMiddlewares(Method.Find)) payload = await middleware[Method.Find](payload);
+
+    payload = await this.provider[Method.Find](payload);
+    payload.trigger = Trigger.PostProvider;
+
+    if (payload.error) throw payload.error;
+
+    for (const middleware of this.middlewares.array()) await middleware.run(payload);
+    for (const middleware of this.middlewares.getPostMiddlewares(Method.Find)) payload = await middleware[Method.Find](payload);
+
+    return payload.data ?? null;
+  }
+
+  /**
+   * Get a value using a key.
+   * @since 2.0.0
+   * @param key A key at which a value is.
+   * @returns The value gotten or null.
+   *
+   * @example
+   * ```typescript
+   * await josh.set('key', 'value');
+   *
+   * await josh.get('key'); // 'value'
+   * ```
+   */
+  public async get(key: string): Promise<StoredValue | null>;
+
+  /**
+   * Get a value using a key and/or path.
+   * @since 2.0.0
+   * @param keyPath A key and/or path at which a value is.
+   * @returns The value gotten or null.
+   *
+   * @example
+   * ```typescript
+   * await josh.set('key', { path: 'value' });
+   *
+   * await josh.get('key'); // { path: 'value' }
+   * await josh.get({ key: 'key' }); // { path: 'value' }
+   * ```
+   *
+   * @example
+   * ```typescript
+   * await josh.set('key', { path: 'value' });
+   *
+   * await josh.get('key.path'); // 'value'
+   * await josh.get({ key: 'key', path: 'path' }); // 'value'
+   * await josh.get({ key: 'key', path: ['path'] }); //'value'
+   * ```
+   */
+  public async get<Value = StoredValue>(keyPath: KeyPath): Promise<Value | null> {
+    const [key, path] = this.getKeyPath(keyPath);
+    let payload: GetPayload<Value> = { method: Method.Get, trigger: Trigger.PreProvider, key, path };
+
+    for (const middleware of this.middlewares.array()) await middleware.run(payload);
+    for (const middleware of this.middlewares.getPreMiddlewares(Method.Get)) payload = await middleware[Method.Get](payload);
+
+    payload = await this.provider.get(payload);
+    payload.trigger = Trigger.PostProvider;
+
+    if (payload.error) throw payload.error;
+
+    for (const middleware of this.middlewares.array()) await middleware.run(payload);
+    for (const middleware of this.middlewares.getPostMiddlewares(Method.Get)) payload = await middleware[Method.Get](payload);
+
+    return payload.data ?? null;
+  }
+
+  /**
+   * Get all stored values.
+   * @since 2.0.0
+   * @param returnBulkType The return bulk type. Defaults to {@link Bulk.Object}
+   * @returns The bulk data.
+   *
+   * @example
+   * ```typescript
+   * await josh.set('key', 'value');
+   *
+   * await josh.getAll(); // { key: 'value' }
+   * await josh.getAll(Bulk.OneDimensionalArray); // ['value']
```

Comment:
Any option that is specifically for TS should be marked as such, as it might confuse non-TS users.

Reply:
This isn't a TypeScript only option. This is to tell Josh how the bulk data should be formatted.